backend/LexBoxApi/GraphQL/LexMutations.cs

```diff
@@ -153,4 +153,16 @@ public async Task<User> ChangeUserAccountByAdmin(ChangeUserAccountByAdminInput i
         await user.ExecuteDeleteAsync();
         return User;
     }
+    [Error<NotFoundException>]
+    [Error<DbError>]
+    [UseMutationConvention]
+    public async Task<User> DeleteUserByUser(DeleteUserByUserInput input, LexBoxDbContext dbContext){
+        var user = await dbContext.Users.FindAsync(input.UserId);
+        if (user is null) throw new NotFoundException("User not found");
+        if (_loggedInContext.User.Id != input.UserId) throw new UnauthorizedAccessException();
+        var User = await dbContext.Users.FindAsync(input.UserId);
```

Comment:
let's try not to have a `User` and `user` variable, that's very confusing. These look like they will be exactly the same anyway so just remove one.

Reply:
Weirdly they don't seem to be the same type? If I switch it to `return user;` instead of `return User`, I get this error:
```red
...LexMutations.cs(152,16): error CS0266: Cannot implicitly convert type 'System.Linq.IQueryable<LexCore.Entities.User>' t 
o 'LexCore.Entities.User'. An explicit conversion exists (are you missing a cast?) [C:\Repos\lexbox\backend\LexBoxApi\LexBoxApi.csproj]
```
should I do something like this? 
`return (LexCore.Entities.User) user;`

Reply:
Looks like you're referring to `DeleteUserByAdmin` as this function was updated just fine. The reason is because on line 151 you are returning a query for a user and not the actual user. Instead of using `ExecuteDeleteAsync` I'd do this:
```c#
dbContext.Users.Remove(user);
dbContext.SaveChangesAsync()
```
similar to what you do with an update.

Reply:
Oh that makes sense, I'll update it, thanks